### PR TITLE
added registers xmm0 to xmm15 while debugging under FreeBSD.

### PR DESCRIPTION
**Pull request by rolio37**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

added xmm0 to xmm15 registers while debugging under FreeBSD.

### CODE REVIEW

File: libr/debug/p/debug_native.c [hunk 1]
```diff
@@ -660,15 +660,24 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
 #endif
 		return true;
 		break;
+
```
Comment (trufae) [inline libr/debug/p/debug_native.c:663]:
```suggestion
```
No empty lines

File: libr/debug/p/debug_native.c [hunk 2]
```diff
@@ -660,15 +660,24 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
...
 	case R_REG_TYPE_VEC512: // ZMM
-		// not implemented
-		break;
+	      #if __KFBSD__
+		struct ptrace_xstate_info info;
+               ret = ptrace(PT_GETXSTATE_INFO, pid, (caddr_t)&info, sizeof(info));
+               
+               if (info.xsave_len != 0)
+               {
+                ret = ptrace(PT_GETXSTATE, pid,  (caddr_t)buf, info.xsave_len);
```
Comment (trufae) [inline libr/debug/p/debug_native.c:677]:
```suggestion
                ret = ptrace (PT_GETXSTATE, pid,  (caddr_t)buf, info.xsave_len);
```

space before (

File: libr/debug/p/debug_native.c [hunk 3]
```diff
@@ -660,15 +660,24 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
...
+               }
+              #endif
```
Comment (trufae) [inline libr/debug/p/debug_native.c:679]:
```suggestion
              #endif
```

dont indent preprocessor directives

File: libr/debug/p/debug_native.c [hunk 4]
```diff
@@ -660,15 +660,24 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
...
+               break;
```
Comment (trufae) [inline libr/debug/p/debug_native.c:680]: do not indent with spaces. We use tabs

File: libr/debug/p/native/bsd/bsd_debug.c [hunk 5]
```diff
@@ -118,8 +118,22 @@ bool bsd_reg_write(RDebug *dbg, int type, const ut8 *buf, int size) {
...
 		break;
+	case R_REG_TYPE_VEC128: // XMM
+	case R_REG_TYPE_VEC256: // YMM
+	case R_REG_TYPE_VEC512: // ZMM
+	        #if __KFBSD__
+		struct ptrace_xstate_info info;
+               r = ptrace(PT_GETXSTATE_INFO, dbg->pid, (caddr_t)&info, sizeof(info));
+               
+               if (info.xsave_len != 0)
+               {
```
Comment (trufae) [inline libr/debug/p/native/bsd/bsd_debug.c:132]: Braces must be opened in the same line as the if

**Review by trufae — CHANGES_REQUESTED** [review]
.

**Comment by trufae** [comment]
Linting not fixed yet. Run the sys/lint.sh script

File: libr/debug/p/debug_native.c [hunk 6]
```diff
@@ -661,13 +661,19 @@ static int bsd_reg_read(RDebug *dbg, int type, ut8* buf, int size) {
...
 	   ret = ptrace (PT_GETFPREGS, pid, (caddr_t)buf, sizeof (struct fpreg));
 	   break;
-	case R_REG_TYPE_VEC64: // MMX
 	case R_REG_TYPE_VEC128: // XMM
 	case R_REG_TYPE_VEC256: // YMM
 	case R_REG_TYPE_VEC512: // ZMM
-		// not implemented
+		#if __KFBSD__
+		struct ptrace_xstate_info info;
+		ret = ptrace(PT_GETXSTATE_INFO, pid, (caddr_t)&info, sizeof (info));
```
Comment (trufae) [inline libr/debug/p/debug_native.c:672]: Space before (

File: libr/debug/p/native/bsd/bsd_debug.c [hunk 7]
```diff
@@ -118,8 +118,20 @@ bool bsd_reg_write(RDebug *dbg, int type, const ut8 *buf, int size) {
...
 	case R_REG_TYPE_FPU:
+	case R_REG_TYPE_VEC64: // MMX
 		r = ptrace (PT_SETFPREGS, dbg->pid, (caddr_t)buf, sizeof (struct fpreg));
 		break;
+	case R_REG_TYPE_VEC128: // XMM
+	case R_REG_TYPE_VEC256: // YMM
+	case R_REG_TYPE_VEC512: // ZMM
+		#if __KFBSD__
+		struct ptrace_xstate_info info;
+		r = ptrace(PT_GETXSTATE_INFO, dbg->pid, (caddr_t)&info, sizeof (info));
```
Comment (trufae) [inline libr/debug/p/native/bsd/bsd_debug.c:129]: Same

File: libr/debug/p/native/bsd/bsd_debug.c [hunk 8]
```diff
@@ -118,8 +118,20 @@ bool bsd_reg_write(RDebug *dbg, int type, const ut8 *buf, int size) {
...
+		if (info.xsave_len != 0) {
+			r = ptrace(PT_SETXSTATE, dbg->pid,  (caddr_t)buf, info.xsave_len);
```
Comment (trufae) [inline libr/debug/p/native/bsd/bsd_debug.c:131]:
```suggestion
			r = ptrace(PT_SETXSTATE, dbg->pid, (caddr_t)buf, info.xsave_len);
```

File: libr/debug/p/native/bsd/bsd_debug.c [hunk 9]
```diff
@@ -118,8 +118,20 @@ bool bsd_reg_write(RDebug *dbg, int type, const ut8 *buf, int size) {
...
+		}
+		#endif
```
Comment (trufae) [inline libr/debug/p/native/bsd/bsd_debug.c:133]: Do not indent this. The # must be at column 0

**Review by trufae — CHANGES_REQUESTED** [review]
.

**Comment by rolio37** [comment]
Hi, thanks for your patience.

**Pull request merged**